### PR TITLE
refactor RenderNotifier into servo/lib.rs

### LINKED ISSUE #22707: Move RenderNotifier from compositing component to servo component
The definition and implementation of the RenderNotifier type is in components/compositing/compositor.rs, but it's only used by components/servo/lib.rs. This is silly; we should extract it so the code that uses it also defines it.

No tests required; if the changes compile, that's good enough for a pull request.

### PR DESCRIPTION
**Pull request by nikopen**
Fixes #22707

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22713)

### CODE REVIEW

**Comment by paulrouget** [comment]
You also need to add the proper `use` in servo/lib.rs, and remove the useless one in compositor.rs.

File: components/servo/lib.rs [hunk 1]
```diff
@@ -132,6 +132,45 @@ pub struct Servo<Window: WindowMethods + 'static> {
     embedder_events: Vec<(Option<BrowserId>, EmbedderMsg)>,
 }
 
+#[derive(Clone)]
+pub struct RenderNotifier {
```
Comment (shanavas786) [inline components/servo/lib.rs]: since its not used anywhere else, we can remove `pub` from here

**Review by shanavas786 — CHANGES_REQUESTED** [review]

**Comment by jdm** [comment]
@nikopen Are you going to finish this work?

**Comment by shanavas786** [comment]
I can take this over if @nikopen is not going to finish.

**Comment by shanavas786** [comment]
I have absorbed this into #22948

**Pull request closed without merging**